internal/util/reader_test.go

```diff
@@ -0,0 +1,77 @@
+package util
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUncompressed(t *testing.T) {
+	// make sure the bytes read are correct for non-gzipped streams
+	nonZipBytes := []byte("this is a test")
+
+	reader, _ := NewReader(io.NopCloser(bytes.NewReader(nonZipBytes)), false, 1000)
+	payloadReader, _ := reader.(*PayloadReader)
```

Comment:
Since we aren't returning the PayloadReader type, we have to do this typehint stuff in a couple of places here.